test/attr/open_objc.swift

```diff
@@ -11,8 +11,14 @@ open class InvalidOpenExtension {
 extension InvalidOpenExtension {
   open func nonObjcOpenMethod() { } // expected-warning {{non-'@objc' methods in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
   open var nonObjcOpenVar: Int { 3 }  // expected-warning {{non-'@objc' properties in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
+  open subscript(_ index: Int) -> Int { 3 } // expected-warning {{non-'@objc' properties in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
+  open convenience init(index: Int) { self.init() } // expected-error {{only classes and overridable class members can be declared 'open'; use 'public'}} // expected-warning {{non-'@objc' methods in extensions cannot be overridden; use 'public' instead}}
+
   @objc open func objcOpenMethod() { } // No warning: @objc methods can be open inside extensions. 
   @objc open var objcOpenVar: Int { 3 } // No warning: @objc methods can be open inside extensions. 
+
+  open class nonObjcClassDecl  { } // No warning about using `open` for nominal types
+  open enum nonObjcEnumDecl { case one } // No warning about using `open` for nominal types // expected-error {{only classes and overridable class members can be declared 'open'; use 'public'}} {{3-7=public}}
```

Comment:
Nit: I know what you're getting at, but the comment is a little confusing, because despite the lack of your specific warning there is another expected error; I think you could just delete the comment and it'd be plenty clear.